debian.sh

```diff
@@ -61,54 +72,63 @@ setup_network_config() {
+    # execute_chroot_command "mdadm --detail --scan | sed -e 's/metadata=00.90/metadata=0.90/g' >> $MDADMCONF"; declare -i EXITCODE=$?
     execute_chroot_command "/usr/share/mdadm/mkconf > $MDADMCONF"; declare -i EXITCODE=$?
+
+    #
     # Enable mdadm
+    #
     sed -i "s/AUTOCHECK=false/AUTOCHECK=true # modified by installimage/" \
-        "$FOLD/hdd/etc/default/mdadm"
+      "$FOLD/hdd/etc/default/mdadm"
     sed -i "s/AUTOSTART=false/AUTOSTART=true # modified by installimage/" \
-        "$FOLD/hdd/etc/default/mdadm"
+      "$FOLD/hdd/etc/default/mdadm"
     sed -i "s/START_DAEMON=false/START_DAEMON=true # modified by installimage/" \
-        "$FOLD/hdd/etc/default/mdadm"
+      "$FOLD/hdd/etc/default/mdadm"
     sed -i -e "s/^INITRDSTART=.*/INITRDSTART='all' # modified by installimage/" \
-        "$FOLD/hdd/etc/default/mdadm"
+      "$FOLD/hdd/etc/default/mdadm"
 
     return "$EXITCODE"
   fi
 }
 
-
 # generate_new_ramdisk "NIL"
 generate_new_ramdisk() {
   if [ -n "$1" ]; then
-    OUTFILE=`ls -1r $FOLD/hdd/boot/initrd.img-* | grep -v ".bak$\|.gz$" | awk -F "/" '{print $NF}' | grep -m1 "initrd"`
-    VERSION=`echo $OUTFILE |cut -d "-" -f2-`
+    OUTFILE=$(find "$FOLD"/hdd/boot -name "initrd.img-*" -not -regex ".*\(gz\|bak\)" -printf "%f\n" | sort -nr | head -n1)
+    VERSION=$(echo "$OUTFILE" |cut -d "-" -f2-)
     echo "Kernel Version found: $VERSION" | debugoutput
 
     if [ "$IMG_VERSION" -ge 60 ]; then
+
+      #
       # blacklist i915 driver due to many bugs and stability issues
+      #
       local blacklist_conf="$FOLD/hdd/etc/modprobe.d/blacklist-hetzner.conf"
-      echo "### $COMPANY - installimage" > "$blacklist_conf"
-      echo '### silence any onboard speaker' >> "$blacklist_conf"
-      echo 'blacklist pcspkr' >> "$blacklist_conf"
-      echo 'blacklist snd_pcsp' >> "$blacklist_conf"
-      echo '### i915 driver blacklisted due to various bugs' >> "$blacklist_conf"
-      echo '### especially in combination with nomodeset' >> "$blacklist_conf"
-      echo 'blacklist i915' >> "$blacklist_conf"
-      echo '### mei driver blacklisted due to serious bugs' >> "$blacklist_conf"
-      echo 'blacklist mei' >> "$blacklist_conf"
-      echo 'blacklist mei-me' >> "$blacklist_conf"
+      {
+        echo "### $COMPANY - installimage"
+        echo "### silence any onboard speaker"
+        echo "blacklist pcspkr"
+        echo "blacklist snd_pcsp"
+        echo "### i915 driver blacklisted due to various bugs"
+        echo "### especially in combination with nomodeset"
+        echo "blacklist i915"
+        echo "### mei driver blacklisted due to serious bugs"
+        echo "blacklist mei"
+        echo "blacklist mei-me"
+      } > "$blacklist_conf"
     fi
 
+    #
     # just make sure that we do not accidentally try to install a bootloader
     # when we haven't configured grub yet
     # Debian won't install a boot loader anyway, but display an error message,
     # that needs to be confirmed
+    #
```

Comment:
There are to kinds of comments, those that contain higher level descriptions should be easily and clearly visible --> Empty comment line before and after.

Commented code lines are without any empty comment lines.